tasks/infra/env/validateZookeeperPath_linux.go

```diff
@@ -0,0 +1,254 @@
+	}
+	/*how we pass the args to the zk shell may differ depending on the version of the shell:
+	zkCli.sh localhost:2181 get /brokers/ids/0
+	vs
+	zookeeper-shell.sh localhost:2181 ls /brokers/ids
+	*/
+	var getArg, brokersArg string
+	if isOlderShellVersion {
+		getArg = "get"
+		brokersArg = defaultZookeeperPath + "/0"
+	} else {
+		getArg = "ls"
+		brokersArg = defaultZookeeperPath
+	}
+
+	hasBrokerList, resultSummary := p.getKafkaBrokersList(zookeeperConfig, zookeeperShellPath, getArg, brokersArg)
+
+	if !hasBrokerList {
+		return tasks.Result{
+			Status:  tasks.Failure,
+			Summary: resultSummary,
+		}
+	}
+
+	return tasks.Result{
+		Status:  tasks.Success,
+		Summary: resultSummary,
+	}
+}
+
+func getZookeeperShellScriptPath(envVars map[string]string) (string, bool) {
+	var path string
+
+	kafkaPath, isKafkaEnvVarPresent := envVars[kafkaEnvVar]
+	zookeeperPath, isZookeeperEnvVarPresent := envVars[zookeeperEnvVar]
+	bashPath, isBashPathPresent := envVars[pathEnvVar]
+
+	if isKafkaEnvVarPresent {
+		//this is our preferred option for finding the path to zookeeper script because is the most popular way to bundle configuration for both kafka and zookeeper
+		path = kafkaPath //Example: KAFKA_HOME=/opt/kafka_2.13-2.6.0
+	} else if isZookeeperEnvVarPresent {
+		path = zookeeperPath //Example: ZOOKEEPER_HOME=C:\Tools\zookeeper-3.4.9
+	} else if isBashPathPresent && strings.Contains(bashPath, "kafka") {
+		/*Examples of paths
+		/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/kafka2.13/bin:/fknvfk
+		/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/opt/something/kafka:/someotherpaths
+		*/
+		regex := regexp.MustCompile(`([^:]+kafka([^:]+|))`)
+		matches := regex.FindStringSubmatch(bashPath)
+		if len(matches) > 0 {
+			path = matches[0]
+		}
+	}
+
+	if len(path) < 1 {
+		return "", false
+	}
+	//It's possible that the path does not include /bin/ (the directory that has zookeeper-shell.sh), in which case we need to add it. Example of full path to the zookeeper shell: /opt/kafka_2.13-2.6.0/bin/zookeeper-shell.sh
+	var binPath string
+	if strings.Contains(path, "bin") {
+		binPath = path
+	} else {
+		binPath = path + "/bin"
+	}
+	//Now verify existing of zookeeper-shell
+	zkFileStatus := tasks.ValidatePath(binPath + "/zookeeper-shell.sh")
+	if zkFileStatus.IsValid {
+		return binPath + "/zookeeper-shell.sh", false
+	}
+	//maybe they have the other less popular version of zookeeper-shell called zkCli.sh
+	olderZkFileStatus := tasks.ValidatePath(binPath + "/zkCli.sh")
+	if olderZkFileStatus.IsValid {
+		return binPath + "/zkCli.sh", true
+	}
+	return "", false
+}
+
+func (p InfraEnvValidateZookeeperPath) getKafkaBrokersList(zookeeperConfig ZookeeperConfig, zookeeperShellPath string, getArg string, brokersArg string) (bool, string) {
+	var hostPortArg string
+	if len(zookeeperConfig.Port) > 0 {
+		hostPortArg = "localhost:" + zookeeperConfig.Port
```

Comment:
Seems we ignore the zookeeper `host` provided in the infra integration config in favor of localhost, curious to the reason we favor that.

Reply:
Good question! I was wondering the same thing about our kafka integration. The integration, as well as the TSEs when they troubleshoot, they completely ignore the host values and troubleshoot around localhost. So I copy the same behavior for nrdiag troubleshooting. I did do a little testing around this, using localhost vs another hostname when running ZK script tool in my command line and it gave me the same output for both. My guess is that somewhere in the ZK implementation this information may already be gathered implicitly or may not be as needed as other pieces of information, such as a port number.